### PR TITLE
add wolfclu enable option and remove test macro guard

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
The removed macro guard was for "./configure --enable-opensslall --disable-ocsp && make check". wolfCLU uses opensslall macros but does not use OCSP.

### CODE REVIEW

**Review by haydenroche5 — CHANGES_REQUESTED** [review]
Suggested a different strategy for the configure.ac changes.

**Comment by kaleb-himes** [comment]
retest this please

File: configure.ac [hunk 1]
```diff
@@ -6412,12 +6407,85 @@ AS_IF([test "x$ENABLED_CMAC" = "xyes" && \
...
+
+if test "$ENABLED_MD5" = "no"
+then
+    AM_CFLAGS="$AM_CFLAGS -DNO_MD5 -DNO_OLD_TLS"
+else
+    # turn off MD5 if leanpsk or leantls on
+    if test "$ENABLED_LEANPSK" = "yes" || test "$ENABLED_LEANTLS" = "yes"
+    then
+        AM_CFLAGS="$AM_CFLAGS -DNO_MD5 -DNO_OLD_TLS"
+        ENABLED_MD5=no
```
Comment (haydenroche5) [inline configure.ac:6471]: Should this be here? I'm under the impression that the `ENABLES` go further up, and the CFLAGS get added down here.

**Review by haydenroche5 — CHANGES_REQUESTED** [review]
Just one question. Otherwise, looks good for final review by PR captain.

**Review by haydenroche5 — APPROVED** [review]
Assigning to @ejohnstown for final review/merge.

**Pull request merged**